Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/di/ProcessLifecycle.kt

```diff
@@ -0,0 +1,8 @@
+package de.rki.coronawarnapp.util.di
+
+import javax.inject.Qualifier
+
+@Qualifier
+@MustBeDocumented
+@Retention(AnnotationRetention.RUNTIME)
+annotation class ProcessLifecycle
```

Comment:
why do we need this?

Reply:
`ProcressLifecycleOwner.get(): LifecycleOwner`, the interface is generic and there are multiple types of `LifecycleOwner`, so they don't get mixed up in the DI graph, we qualify this specific type of `LifecycleOwner` similar to how we handle `Context`.

Reply:
```kotlin
class Example constructor(@ProcessLifecycle val lifecycleOwner: LifecycleOwner)
```